From 03b605a70c8f5a6def7a572ceb302051934c78e7 Mon Sep 17 00:00:00 2001
From: Stanislav Laznicka <slaznick@redhat.com>
Date: Fri, 21 Apr 2017 09:32:34 +0200
Subject: [PATCH] compat-manage: behave the same for all users

Due to LDAP connection refactoring, compat-manage would have behaved
differently for root and for other users even though it requires
the directory manager password. This is caused by it trying to do
external bind when it does not have the DIRMAN password which was
previously not supplied.

https://pagure.io/freeipa/issue/6821

Reviewed-By: Martin Basti <mbasti@redhat.com>
---
 install/tools/ipa-compat-manage | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/install/tools/ipa-compat-manage b/install/tools/ipa-compat-manage
index a29a92fabd653bb983778420cca0983048ee39ef..6dd259d2aad870e575093d6732157de743502ac2 100755
--- a/install/tools/ipa-compat-manage
+++ b/install/tools/ipa-compat-manage
@@ -105,7 +105,7 @@ def main():
                   debug=options.debug,
                   confdir=paths.ETC_IPA)
     api.finalize()
-    api.Backend.ldap2.connect()
+    api.Backend.ldap2.connect(bind_pw=dirman_password)
 
     if args[0] == "status":
         entry = None
-- 
2.12.2

